### ISSUE TITLE
Add support for Makefile as build script and make as build tool

### ISSUE THREAD

**Issue by behnazh-w** [oracle/macaron#833]
Makefile is often used as a generic build tool. In fact, [Macaron itself ](https://github.com/oracle/macaron/blob/main/Makefile)uses `Makefile` to generate artifacts. We already have support for [a number of build tools](https://oracle.github.io/macaron/pages/supported_technologies/index.html#build-tools). 

To support `Makefile` 

- we need to add a parser for `Makefile` to extract shell commands, similar to our [bashparser.](https://github.com/oracle/macaron/blob/5c6c9138fbbc01c19faaf8bcadf0e5cc09ba1149/src/macaron/parsers/bashparser.py)
- add a new module for `make` similar to other [build tools](https://github.com/oracle/macaron/tree/5c6c9138fbbc01c19faaf8bcadf0e5cc09ba1149/src/macaron/slsa_analyzer/build_tool).

**Comment by jenstroeger** [comment]
I poked around a little, and here are a few references. First off, _It would be tricky to write a grammar for make, since the grammar is extremely context-dependent._ ([thread](https://www.mail-archive.com/help-make@gnu.org/msg02773.html), similar questions [here](https://www.mail-archive.com/help-make@gnu.org/msg05999.html) or especially [here](https://www.mail-archive.com/help-make@gnu.org/msg13362.html)).

Having said that, I found the following attempts to come up with a Makefile grammar:
- https://github.com/Icraus/MakefileGrammar
- https://github.com/grammarjs/makefile (requires other repos from the same org)

It might also make sense to explore existing syntax highlighters like [Pygments](https://github.com/pygments/pygments) which support at least [Makefile lexers](https://pygments.org/docs/lexers/#lexers-for-makefiles-and-similar).

@behnazh-w given a Makefile, it looks like you’re interested in extracting the [Recipes](https://www.gnu.org/software/make/manual/html_node/Recipes.html) from it?

**Comment by behnazh-w** [comment]
> @behnazh-w given a Makefile, it looks like you’re interested in extracting the [Recipes](https://www.gnu.org/software/make/manual/html_node/Recipes.html) from it?

Yes exactly. Our ultimate goal is to find shell commands that invoke build tools. For now we can limit the Recipes to bash scripts.
